### PR TITLE
[Architecture] AccountsTab

### PR DESCRIPTION
**Pull request by Doomsdayrs**
### Description of the Change

While maintaining as much of the current code form as possible, Move data to a view model.

### Testing

- Microsoft account
  - [x] Login
  - [x] Relogin
  - [x] Delete
  - [ ] Update username(?)
    - Fails
  - [x] Reload skin
    - Seems to work
  - [x] Refresh access token
- Mojang account
  - Unable to test, lacking legacy account 

### Related Issues

#606

### CODE REVIEW

**Comment by Doomsdayrs** [comment]
@RyanTheAllmighty I am about done with this, I want to test it before I make it not WIP.

How do I test a non-Microsoft account?

**Comment by RyanTheAllmighty** [comment]
Unless you have one, it's impossible to test unfortunately. Testing with an invalid login as well as if you have a migrated account that used to be a Mojang account  so it shows that it's migrated is probably fine enough

**Comment by Doomsdayrs** [comment]
Alright, I can test again for sanity, but the login via Microsoft works fine.

For legacy accounts? 50/50. I'll test to see if it says "migrated", otherwise we would have to see user complaints come in. Unless you want to ping your members to see if anyone with a legacy account can help test?

**Comment by RyanTheAllmighty** [comment]
Nah as far as I know nobody should be using them anymore, I see more and more people just being blocked from logins now. If there's an issue it should be obvious in the code or if not someone will complain and can fix it then. But once out of draft I'll take a look

**Comment by Doomsdayrs** [comment]
@RyanTheAllmighty 

If you would like to test it once more, read the code to make sure the logic checks out (I am PS it does).
Afterwards, A monumental step forward to making the app easier to code.

**Comment by Doomsdayrs** [comment]
@RyanTheAllmighty This PR will break UI test, as the UI test users `setText` instead of actually typing in the text

**Comment by RyanTheAllmighty** [comment]
> @RyanTheAllmighty This PR will break UI test, as the UI test uses `setText` instead of actually typing in the text

This fixes it

https://github.com/ATLauncher/ATLauncher/blob/eeb185596ee56bc98c3d03e953c9d6264739a52f/src/test/java/ui/BasicLauncherUiTest.java#L78-L79

```java
        usernameField.setText("test@example.co");
        usernameField.pressAndReleaseKey(KeyPressInfo.keyCode(KeyEvent.VK_M));
        passwordField.setText("passwor");
        passwordField.pressAndReleaseKey(KeyPressInfo.keyCode(KeyEvent.VK_D));
```

**Comment by Doomsdayrs** [comment]
Alright, With that done the CI should check out.

Also, If you may, Please use a squash commit for this PR merge, to make it very easy to rollback later.

**Comment by RyanTheAllmighty** [comment]
@Doomsdayrs looks like username change in the context menu is broken. Just does nothing when clicked. Looks working in latest release

File: src/main/java/com/atlauncher/gui/tabs/accounts/AccountsTab.java [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+
+            // Have this done via listener
+            AbstractAccount account = viewModel.getSelectedAccount();
+            userSkin.setIcon(account.getMinecraftSkin());
+        });
+        contextMenu.add(updateSkin);
+
+        JMenuItem updateUsername =
+            new JMenuItem(GetText.tr("Update Username"));
+        updateUsername.addActionListener(e -> viewModel.updateUsername());
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/gui/tabs/accounts/AccountsTab.java:298]: Looks like this doesn't work. Just does nothing when clicked. Looks working in latest release

File: src/main/java/com/atlauncher/gui/tabs/accounts/AccountsTab.java [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+        rightPanel.add(topPanel, BorderLayout.NORTH);
+        rightPanel.add(bottomPanel, BorderLayout.CENTER);
+
+        contextMenu = new JPopupMenu();
+
+        updateSkin = new JMenuItem(GetText.tr("Reload Skin"));
+        updateSkin.addActionListener(e -> {
+            viewModel.updateSkin();
+
+            // Have this done via listener
```
Comment (RyanTheAllmighty) [inline src/main/java/com/atlauncher/gui/tabs/accounts/AccountsTab.java:290]: If this is a TODO best to mark it as one

**Review by RyanTheAllmighty — CHANGES_REQUESTED** [review]

File: src/main/java/com/atlauncher/gui/tabs/accounts/AccountsTab.java (hunk 1 above)
Comment (Doomsdayrs) [reply src/main/java/com/atlauncher/gui/tabs/accounts/AccountsTab.java:299]: I believe during development I thought this was something for mojang accounts. I was wrong, thus I had an instance check. I fixed that with [92372c1](https://github.com/ATLauncher/ATLauncher/pull/607/commits/92372c1862c5b751922a791fa0038a785ebc7f4c).

File: src/main/java/com/atlauncher/gui/tabs/accounts/AccountsTab.java (hunk 2 above)
Comment (Doomsdayrs) [reply src/main/java/com/atlauncher/gui/tabs/accounts/AccountsTab.java]: Fixed with [d781719](https://github.com/ATLauncher/ATLauncher/pull/607/commits/d78171975999425a923b91e5c8f21a777bfff852) and also further detailed what I mean.

**Review by RyanTheAllmighty — APPROVED** [review]

**Pull request merged**